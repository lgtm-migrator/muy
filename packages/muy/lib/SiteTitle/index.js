import React from "react"
import GatsbyLink from "../GatsbyLink"
import { H1 } from "../Heading"

const PureSiteTitle = ({ color, children, HeadingProps, ...props }, ref) => (
  <H1 gutterBottom={false} variant={"body1"} {...HeadingProps}>
    <GatsbyLink
      color={"inherit"}
      ref={ref}
      to={"/"}
      underline={"hover"}
      {...props}
    >
      {children}
    </GatsbyLink>
  </H1>
)

const SiteTitle = React.forwardRef(PureSiteTitle)

export default SiteTitle

SiteTitle.defaultProps = {
  color: "inherit",
}
