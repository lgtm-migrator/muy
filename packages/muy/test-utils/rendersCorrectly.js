import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
