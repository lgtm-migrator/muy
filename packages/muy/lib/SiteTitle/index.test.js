import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import SiteTitle from "./index"

describe("<SiteTitle />", () => {
  it("should render correctly", () => {
    rendersCorrectly(<SiteTitle />)
  })
})
