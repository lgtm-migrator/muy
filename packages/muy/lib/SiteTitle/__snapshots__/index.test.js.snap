// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteTitle /> should render correctly 1`] = `
<h1
  className="MuiTypography-root MuiTypography-body1"
>
  <a
    aria-current="page"
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
    target="_self"
  >
    Test site title
  </a>
</h1>
`;

exports[`<SiteTitle /> should render correctly with inherit color 1`] = `
<h1
  className="MuiTypography-root MuiTypography-body1"
>
  <a
    aria-current="page"
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
    target="_self"
  />
</h1>
`;

exports[`<SiteTitle /> should render correctly with primary color 1`] = `
<h1
  className="MuiTypography-root MuiTypography-body1"
>
  <a
    aria-current="page"
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
    target="_self"
  />
</h1>
`;
